### PR TITLE
[#1612] Wire dead /g/:slug/system to real GroupSettingsPage; ?g= for active-group context

### LINKED ISSUE #1612: [FE] Sidebar "Settings" dead-ends on placeholder labelled #1414; active group dropped on path-clean routes
## Symptoms

Two related symptoms surfaced while testing #1604:

### 1. Sidebar "Settings" leads to a dead placeholder

Sidebar's MANAGE.system entry (`common:nav.system` → labelled "Settings") routes to `/g/:slug/system`, which renders `<PlaceholderPage trackedBy="#1414" />` — i.e. "**System** Coming soon. Tracked by #1414".

But:

- **#1414** is `[FE] Profile + User Settings (theme, density, locale, currency, sign-out, danger zone)` — **user-level**, already shipped at `/profile` and `/settings` (see `AppSidebar.tsx` Personal section).
- The real **group settings page** (issue #1413, also shipped) lives at `/groups/:groupId/settings` — `frontend/src/pages/groups/GroupSettingsPage.tsx`. After #1604 (currency migration PR 4/4) it carries the migrate-currency wizard, the lock UX banner, the migrations-history Sheet, and the danger zone.
- Nothing in the sidebar links to `/groups/:id/settings`. The only entry point was the GroupSelector dropdown's "Group settings" item.

So clicking the sidebar's Settings sends the user into a stale stub mislabelled with the wrong issue, while the real page sits at a URL the navigation never references.

### 2. Sidebar collapses on `/profile`, `/settings`, `/groups/:id/settings`

`AppSidebar.tsx` resolved active-group via `useParams<{ groupSlug?: string }>()`, which is non-null only on `/g/:groupSlug/*`. On any path-clean route — Profile, Preferences, GroupSettings — `groupSlug` is `null`, every Inventory + Manage `NavEntry.to(null)` returns `null`, and `NavRow` renders nothing for them. Only the Personal section remains visible.

The intent in `GroupContext.tsx:39-40` was *"URL is the single source of truth"* — strictly path-driven. Consequence: a user with one or more groups loses their group-aware nav the moment they navigate off `/g/:slug/*`, even though they unambiguously have an active group.

## Root principle

Non-group routes are only legal when the user has **zero** groups. Once they have ≥1 group, every URL must encode the active group somewhere.

Migrating every URL into `/g/:slug/*` (e.g. `/profile` → `/g/:slug/profile`) was rejected — paths get longer, refactor is invasive, and there's no good URL shape for create-first-group. Instead: **add `?g=<slug>` query param** for group context on path-clean routes. The path stays untouched; the sidebar (and anything else reading currentGroup) gets a coherent active group.

### Wire dead routes to real implementations

- `AppSidebar.tsx` MANAGE.system → `/groups/${currentGroup.id}/settings`. Switch `NavEntry.to` signature from `(slug)` to `(group: LocationGroup | null)` so each entry uses what it needs (slug for `/g/:slug/*`, id for `/groups/:id/*`, no group for Personal).
- `router.tsx`: drop both `<Route path="system">` placeholders.
- `stubs.json`: drop `system` and `systemSetting` keys.

### Introduce `?g=<slug>` for active-group context on path-clean routes

- `GroupContext.tsx` resolves currentGroup from path `:groupSlug` → path `:groupId` → query `?g=<slug>`. Auto-pins `?g=<defaultSlug>` (replace) on a stable allowlist of routes (`/profile`, `/profile/edit`, `/settings`, `/no-group`, `/groups/new`, `/plans`, `/help`, `/help/shortcuts`, `/whats-new`) when no other group hint is present. Excluded explicitly: redirect-target stubs (`/locations`, `/files`, `/system`, …) — `UngroupedRedirect` preserves search/hash on rewrite, so adding `?g=` there leaks into the redirected URL.
- New helper `frontend/src/lib/group-aware-url.ts` → `withGroupQuery(path, slug)`.
- Outgoing links to path-clean routes carry `?g=` via the helper: AppSidebar Personal entries + Profile dropdown, GroupSelector "Create group", ProfilePage (→ /settings, /profile/edit, /plans), EditProfilePage (→ /profile), SettingsPage (→ /profile/edit).
- AppSidebar reads `currentGroup` from `useCurrentGroup()` instead of `useParams`, so the inventory/manage rows render consistently regardless of which URL surface the user is on.

## Acceptance criteria

- [ ] Sidebar's "Settings" entry navigates to the real `GroupSettingsPage` at `/groups/:id/settings`.
- [ ] On `/profile`, `/settings`, `/groups/:id/settings`, the sidebar shows full Inventory + Manage rows; URL carries `?g=<active-slug>` (auto-set or carried).
- [ ] Direct entry / bookmark to `/profile` (no `?g=`) for a user with groups: GroupContext auto-pins `?g=<default-slug>` and the sidebar populates.
- [ ] `/g/:slug/*` URLs unchanged (no `?g=` injected).
- [ ] `UngroupedRedirect` paths (`/files`, `/locations`, …) are NOT mutated by GroupContext (no `?g=` leak in the redirected URL).
- [ ] Placeholder routes for `/g/:slug/system*` removed; i18n stubs `system`/`systemSetting` gone.
- [ ] Dead `PlaceholderPage` import gone from `router.tsx`.
- [ ] `npm run typecheck` clean, `npm run lint` 0 errors, `npm run test` all passing, `npm run format:check` clean.

## Out of scope (follow-up)

- TopBar account dropdown / other shells linking to non-group routes. Audit on next touch.
- The `/g/:slug/*` vs `/groups/:id/*` URL-shape drift itself. Both forms remain; admin pages (settings, members) keep id-based, navigation pages keep slug-based. The id-based form is reachable via GroupContext's id-fallback so the principle still holds.
- Migrating `/groups/new` to a group-aware URL when the user has ≥1 groups — currently mitigated by pinning `?g=` via the helper; full path migration deferred.

### PR DESCRIPTION
**Pull request by denisvmedia**
Closes #1612.

## Summary

Two related bugs, one PR.

### 1. Sidebar "Settings" was a dead placeholder mislabeled with #1414

`MANAGE.system` (sidebar "Settings") routed to `/g/:slug/system`, a `<PlaceholderPage trackedBy="#1414" />`. But #1414 is the user-level Profile + Settings (already shipped at `/profile`, `/settings`); the real **group** settings page lives at `/groups/:id/settings` (issue #1413, also shipped, and host of the #1604 currency-migration entry-point). Nothing in navigation pointed there — only the GroupSelector dropdown.

Fix: repoint `MANAGE.system` to `/groups/${group.id}/settings`. Drop both `/g/:slug/system*` placeholder routes and their i18n stubs (`system`, `systemSetting`). Drop the now-unused `PlaceholderPage` import from `router.tsx`.

### 2. Sidebar collapsed on path-clean routes (`/profile`, `/settings`, `/groups/:id/settings`)

`AppSidebar` resolved the active group from `useParams<{ groupSlug?: string }>()` only. Outside `/g/:slug/*` the slug is `null`, every Inventory + Manage `NavEntry.to(null)` returns `null`, and `NavRow` renders nothing for them — leaving only Personal visible.

Project principle (per maintainer): non-group routes are only legal for users with **zero** groups. Once they have ≥1 group, every URL must encode the active group somewhere. Migrating every URL into `/g/:slug/*` was rejected; the group context now piggybacks on the URL via a **`?g=<slug>` query param**, leaving paths untouched.

`GroupContext` changes:
- Resolve `currentGroup` from path `:groupSlug` → path `:groupId` → query `?g=<slug>`.
- Auto-pin `?g=<defaultSlug>` (replace) on a stable allowlist of routes when missing — `/profile`, `/profile/edit`, `/settings`, `/groups/new`, `/plans`, `/help`, `/help/shortcuts`, `/whats-new`. Default = `user.default_group_id` → first slug-bearing group.
- **Allowlist excludes `UngroupedRedirect` targets** (`/locations`, `/files`, …) — that component preserves search/hash on rewrite, and adding `?g=` there leaks into the redirected URL. **Also excludes `/no-group`** — it's the zero-group state, no active group to pin, and the transient post-create case (a fresh group surfacing while still on `/no-group` before `navigate("/")` fires) races the inline auto-set on firefox/webkit and would otherwise leave the user stuck on `/no-group?g=…`.
- HTTP rewrite slot synced from URL hint immediately (path slug ?? query slug); corrected from `currentGroup.slug` once `/groups/:id` resolves. Prevents the null-window where group-scoped fetches fire un-rewritten on first paint.
- Tolerates missing `<AuthProvider>` via `useOptionalAuth`, so test mounts without auth keep working.

`AppSidebar`:
- `NavEntry.to` signature changed from `(slug)` to `(group: LocationGroup | null)` so each entry uses what it needs (slug for `/g/:slug/*`, id for `/groups/:id/*`, no group for Personal).
- Reads `currentGroup` from `useCurrentGroup()` instead of `useParams`. Inventory + Manage rows render coherently regardless of URL shape.
- Personal entries + Profile dropdown link pin `?g=<slug>` via `withGroupQuery`.

Outgoing-link sites updated for path-clean targets (helper: `frontend/src/lib/group-aware-url.ts → withGroupQuery(path, slug)`):
- `GroupSelector`: "Create group" navigation.
- `ProfilePage`: `/settings`, `/profile/edit`, `/plans` links.
- `EditProfilePage`: `/profile` back/cancel links.
- `SettingsPage`: `/profile/edit` edit-profile + change-password CTAs.
- `CommandPalette`: same `(group)` signature switch + `?g=` on Personal entries.

`/system` legacy alias removed entirely (no SystemRedirect, no UngroupedRedirect mapping). The only canonical group-settings URL is now `/groups/:id/settings`.

## ⚠️ Operator-visible behavior change

Cleaning up an unrelated footgun the e2e flake exposed: the `FeatureCurrencyMigration` flag (gate for the `/currency-migrations` endpoints, lock middleware, and worker — shipped under #1604) defaulted to `false` in the binary, so `inventario run` without `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION=true` rendered the wizard inert and 404'd the endpoints. CI / docker-compose / setup-stack worked around it by always setting the env to `true`; that workaround was a smell.

This PR ships:

| Surface | Change |
| --- | --- |
| `go/cmd/inventario/run/bootstrap/config.go` | `FeatureCurrencyMigration` `env-default` flipped from `"false"` to `"true"`. The flag becomes a kill-switch (set to `false` to disable) instead of a launch-gate (must opt in). Comment + `apiserver.go` `Params` doc updated. |
| `helm/inventario/values.yaml` | New top-level `features:` block with `currencyMigration: true` — operator-facing toggle, documented as the canonical surface for future feature flags. |
| `helm/inventario/templates/configmap.yaml` | Wires `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION` from `.Values.features.currencyMigration`. ConfigMap is consumed via `envFrom` on every deployment in the chart (web / apiserver / workers / setup job). |
| `k8s/dev/inventario/configmap.yaml`, `k8s/prod/configmap.yaml` | Mirror the Helm `features:` block — `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION: "true"` explicit so an operator running raw `kubectl apply -f k8s/{dev,prod}/` has an obvious place to flip the kill-switch. |
| Workaround removals | `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION=true` overrides dropped from `docker-compose.e2e.yaml`, `.github/workflows/e2e-tests.yml` (macOS webkit lane), and `e2e/setup/setup-stack.ts` (operator-override hatch via `process.env` preserved). |

**Upgrade impact:** operators who explicitly want the surface dark must now set `features.currencyMigration: false` (Helm) or `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION: "false"` (raw env). Existing deployments that already set the env to `true` continue to work unchanged.

## Visual proof

See [the issue comment](https://github.com/denisvmedia/inventario/issues/1612#issuecomment-4412142739) for screenshots — `/profile`, `/settings`, `/groups/:id/settings` all render the full sidebar; `/g/:slug/` dashboard unchanged.

## Out of scope

- TopBar / other shells linking to non-group routes — audit on next touch.
- Full migration of `/profile`, `/settings`, `/groups/new` paths into a single URL shape (rejected by maintainer in favor of `?g=`).
- The `/g/:slug/*` vs `/groups/:id/*` URL-shape drift — both forms remain; the principle holds because `idFromPath` resolution makes `/groups/:id/*` group-aware.

## Test plan

- [x] `npm run typecheck` — clean
- [x] `npm run lint` — 0 errors (10–11 pre-existing react-compiler warnings on `form.watch()`, untouched)
- [x] `npm run format:check` — clean
- [x] `npm run test` — 740/740 passing (incl. new GroupContext + withGroupQuery cases)
- [x] `go test ./cmd/... ./apiserver/...` — clean
- [x] Local e2e: `navigation.spec.ts`, `groups.spec.ts`, `delete-group-dialog.spec.ts`, `currency-migration.spec.ts`, `profile.spec.ts` all chromium-passing (admin fixture)
- [ ] Postgres e2e (CI lane) — ?g= auto-set fix on `/no-group` removal verified across firefox/webkit + chromium

### CODE REVIEW

File: frontend/src/lib/group-aware-url.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// (/profile, /settings, /groups/new, …) — see GroupContext for how it's
+// resolved. The path-prefix shape `/g/:slug/*` and `/groups/:id/*` win
+// over `?g=`; on those routes the query is redundant and may be omitted.
+//
+// Returns the original `path` when `slug` is falsy (anonymous user, or
+// user-with-zero-groups state) so the link still resolves.
+export function withGroupQuery(path: string, slug: string | null | undefined): string {
+  if (!slug) return path
+  const sep = path.includes("?") ? "&" : "?"
+  return `${path}${sep}g=${encodeURIComponent(slug)}`
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/lib/group-aware-url.ts]: Good catch — fixed in 8b83979.

`withGroupQuery` now slices off any `#fragment` first, parses the query via `URLSearchParams` (so `set("g", slug)` replaces an existing `g=` instead of appending), then reattaches the fragment at the tail. 7 new vitest cases in `frontend/src/lib/__tests__/group-aware-url.test.ts` lock the behavior in:

- empty / null slug → unchanged
- clean path → `?g=…`
- existing query → merge before the new pair
- existing `g=` → replace, no duplicate
- hash-only path → `?g=…` lands before the hash
- query + hash → both preserved, `g=` merged
- URL-special chars in slug → properly encoded

In production all three call sites pass plain alphanumeric paths from the i18n nav config, so neither bug fires today, but it's the kind of helper I'd expect future call sites (e.g. tying ?g= onto a deep-link) to misuse — locking it down now.

File: frontend/src/components/AppSidebar.tsx [hunk 2]
```diff
@@ -44,101 +44,120 @@ import {
...
+    // identity, currency, danger zone. Used to point at a /g/:slug/system
+    // placeholder mislabeled with #1414 (which actually shipped the
+    // user-level Profile + Settings, distinct surface). Path uses :id
+    // because that's where the page is mounted today; pinning ?g= would
+    // be redundant since the id alone resolves the group via
+    // GroupContext's id-fallback.
     labelKey: "common:nav.system",
-    to: (slug) => (slug ? `/g/${encodeURIComponent(slug)}/system` : null),
+    to: (g) => (g?.id ? `/groups/${encodeURIComponent(g.id)}/settings` : null),
     icon: Settings,
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/components/AppSidebar.tsx:137]: Fixed in 8b83979. CommandPalette had the same `(slug) => string | null` shape AppSidebar.NavEntry used to have — so the `nav.system` arm still pointed Cmd/Ctrl+K at the now-deleted `/g/:slug/system` placeholder, exactly the dead-end you flagged.

Switched `PaletteEntry.to` to `(group: LocationGroup | null) => string | null` to match `AppSidebar.NavEntry`, and read `currentGroup` from `useCurrentGroup()` (was `useParams`) so the palette stays consistent regardless of URL shape. The `nav.system` arm now points at `/groups/${currentGroup.id}/settings`, mirroring the sidebar; `nav.profile` / `nav.preferences` pin `?g=<slug>` via `withGroupQuery` for the same reason the sidebar Personal entries do.

File: frontend/src/features/group/__tests__/GroupContext.test.tsx [hunk 3]
```diff
@@ -101,8 +161,28 @@ describe("GroupContext", () => {
     await waitFor(() =>
       expect(screen.getByTestId("probe").getAttribute("data-groups")).toBe("household,office")
     )
-    expect(screen.getByTestId("probe").getAttribute("data-current")).toBe("")
-    expect(getCurrentGroupSlug()).toBeNull()
+    expect(screen.getByTestId("loc").getAttribute("data-search") ?? "").not.toContain("g=")
+  })
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/features/group/__tests__/GroupContext.test.tsx:174]: Fair — fixed in 8b83979.

Added the two missing assertions so the test now matches its title: `data-current=""` (currentGroup stayed null) and `getCurrentGroupSlug() === null` (http rewrite slot stayed null) — both proving GroupContext didn't quietly resolve a group via some other branch. The existing `data-search` assertion still guards against the original concern (no `?g=` leak into a UngroupedRedirect rewrite). Comment block above the assertions now spells out all three invariants the case is protecting.

**Comment by denisvmedia** [comment]
Pushed 8b83979 with the three Copilot review fixes (replies threaded inline above):

| Thread | Fix |
| --- | --- |
| `withGroupQuery` hash + dup-`g=` | Hash-aware split + `URLSearchParams.set` so `g=` replaces, `?` stays before `#`. Locked in by 7 new vitest cases. |
| CommandPalette dead-link | Same `(slug) → (group)` signature switch as AppSidebar; `nav.system` now points at `/groups/:id/settings`, profile/preferences pin `?g=`. |
| Test title vs assertions | Added `data-current === ""` + `getCurrentGroupSlug() === null` assertions; comment block spells out all three invariants the case protects. |

Local checks: `typecheck` ✓ · `test` 727/727 (was 720; +7 new helper cases) · `lint` 0 errors (10 pre-existing `form.watch()` warnings) · `format:check` ✓.

### E2E status

Re-ran the previous failure set (`groups.spec.ts`, `delete-group-dialog.spec.ts`, `profile.spec.ts`, `currency-migration.spec.ts`) locally against this branch's frontend (vite mode-C) with `INVENTARIO_RUN_MAX_GROUP_MEMBERSHIPS_PER_USER=0` and a fresh seeded backend — **all 38 tests pass**. The fix doesn't appear to break any path I can hit with the seeded admin fixture.

The only spec I couldn't reproduce locally is `no-group-redirect.spec.ts` and the orphan branch of `settings-default-group.spec.ts`, because the `orphan@test-org.com` zero-group fixture isn't creatable on a memory:// backend (the `users create` CLI is postgres-only, and tenant `test-org` has registration closed). CI provisions orphan via a dedicated workflow step, so the new run on origin should exercise that branch with the corrected GroupContext + sidebar wiring; the auto-set effect bails on `groups.length === 0` so orphan flows shouldn't be touched, but the watch is on this iteration of CI to confirm.

The `currency-migration.spec.ts:77` "wizard preview surfaces totals" timeout reproduces here too on the same `[data-testid="wizard-total-before"]` selector — same flake the screenshot pass hit on master right after pulling #1604; doesn't appear PR-specific. Worth a separate stability pass on the wizard's step 3 if it persists in the next run.

**Comment by denisvmedia** [comment]
Update on the `currency-migration.spec.ts:77` "wizard preview surfaces totals" timeout — it wasn't a flake. Root cause: the binary defaulted `FeatureCurrencyMigration` to `false`, so any `inventario run` without `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION=true` returned 404 from the preview endpoint and step 3 of the wizard never rendered. Local repro nailed it after I started the binary with the flag explicit; the test passed without further changes.

The PR description for #1604 (where the flag landed) said the surface "is end-to-end usable in staging" — that's only true when each operator manually wires the env. This is the kill-switch-shipped-as-launch-gate footgun.

Pushed 23b0c69 to fix it properly:

- **`helm/inventario/values.yaml`** — new top-level `features:` block with `currencyMigration: true`. Operator-facing toggle, documented as the canonical surface for future feature flags.
- **`helm/inventario/templates/configmap.yaml`** — wire the value to `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION`. The configmap is consumed via `envFrom` on every deployment in the chart (web / apiserver / workers / setup job), so a single line covers the whole pod set.
- **`go/cmd/inventario/run/bootstrap/config.go`** — flip `env-default` from `"false"` to `"true"` so the binary's intrinsic default matches the Helm default. `inventario run` with no env now produces the same surface as a fresh `helm install`. Comment + `apiserver.go` `Params` doc updated to describe the flag as a kill-switch rather than a launch gate.

Verified e2e end-to-end: `curl … /api/v1/seed` → `npx playwright test --project=chromium tests/currency-migration.spec.ts -j 1` against the rebuilt binary with **no explicit env override** → 2/2 passing in 8.3s.

Same path drops the `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION=true` workaround from CI's macOS lane (workflow file) and the docker-compose / setup-stack files — those become no-ops once #23b0c69 lands. Cleaning them up is a trivial follow-up; left for now to keep this PR focused on the e2e flake fix.

**Comment by denisvmedia** [comment]
Pushed da78337 to fold the cleanup into this PR (no follow-up):

**Dropped now-redundant `FEATURE_CURRENCY_MIGRATION=true` overrides** (binary default = true since 23b0c69, so each was a no-op):

- `docker-compose.e2e.yaml`
- `.github/workflows/e2e-tests.yml` (macOS webkit lane)
- `e2e/setup/setup-stack.ts` — kept the operator-override hatch (passes `process.env.INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION` through when set, otherwise lets the binary default handle it)

**Mirrored Helm `features:` into raw-YAML alts** (per "тоже надо выставить секцию features"):

- `k8s/dev/inventario/configmap.yaml`
- `k8s/prod/configmap.yaml`

Both get `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION: "true"` explicitly. The binary default agrees, but listing the flag in the configmap gives an operator running raw `kubectl apply -f k8s/{dev,prod}/` an obvious place to flip the kill-switch without grepping for env vars — same affordance the Helm chart's `features:` block provides.

Re-verified: rebuilt binary, restarted with neither `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION` nor `INVENTARIO_RUN_MAX_GROUP_MEMBERSHIPS_PER_USER=0` (the latter still required for shared-admin tests, kept), `npx playwright test --project=chromium tests/currency-migration.spec.ts -j 1` → 2/2 passing in 8.5s. CI will exercise the same default path on the next run.

**Comment by denisvmedia** [comment]
CI on da78337 was down to **one** failing chromium test (Playwright report attached to the run, downloaded + decoded — all the other 25 failures from the original run were the cap-and-flag infrastructure issues that the previous commits handled):

```
navigation.spec.ts > Application Navigation > should navigate to system page
  Error: Failed to authenticate - login did not complete successfully
```

Root cause: the test does `navigateWithAuth(page, '/system')`, the e2e helper's `gotoScoped` rewrote that to `/g/<slug>/system`, and the placeholder route at that path is the one #1612 deleted. So the catch-all `NotFoundPage` rendered **outside** the auth Shell, the `[data-testid="user-menu"]` probe `isAuthenticated` uses came back empty, and the helper concluded "not authenticated" and threw.

Pushed a08e86f:

- **`frontend/src/components/routing/SystemRedirect.tsx`** (new) — mirrors `UngroupedRedirect`'s group-resolution but bounces to `/groups/<active-id>/settings` (the real `GroupSettingsPage` from #1413). Zero-group users still land on `/no-group`, same behavior as before, so `no-group-redirect.spec.ts`'s `/system` case keeps passing.
- **`frontend/src/app/router.tsx`** — `/system` now mounts `SystemRedirect` instead of `UngroupedRedirect`. Dropped `/system/*` (no real-page equivalent for the deleted `/g/:slug/system/settings/:id` placeholder; sub-paths fall through to the catch-all, consistent with the rest of the dead /:slug/* paths).
- **`e2e/tests/includes/group-url.ts`** — `/system` removed from `FLAT_DATA_PREFIXES`; SystemRedirect handles the rewrite SPA-side now, so `gotoScoped` must let `/system` through verbatim.
- **`e2e/tests/navigation.spec.ts`** — renamed the test and updated the assertions to match the new destination: URL ends with `/groups/<id>/settings`, page renders `[data-testid="group-settings-page"]`.

Local verification: `navigation.spec.ts` 4/4 chromium passing in 18s. Full unit suite 727/727. Typecheck + lint (0 errors) + format all clean.

**Comment by denisvmedia** [comment]
51a16b6 reverses the SystemRedirect alias from a08e86f and rips `/system` out entirely (per "no legacy"):

- `frontend/src/components/routing/SystemRedirect.tsx` — deleted.
- `frontend/src/app/router.tsx` — `/system` route gone, no redirect at all.
- `frontend/src/components/routing/UngroupedRedirect.tsx` doc-comment — `/system` removed from the legacy-paths list.
- `frontend/src/components/AppSidebar.tsx`, `frontend/src/components/CommandPalette.tsx` — historical references to the deleted `/g/:slug/system` placeholder trimmed from comments.
- `e2e/tests/navigation.spec.ts` — "should navigate to system page" test deleted (target URL doesn't exist; coverage of the destination is already in delete-group-dialog / currency-migration / groups specs).
- `e2e/tests/no-group-redirect.spec.ts` — `/system` removed from the iteration list.
- `e2e/tests/includes/navigate.ts` — `TO_SYSTEM` / `FROM_SYSTEM` constants and the `case TO_SYSTEM` branch dropped.
- `e2e/tests/includes/group-url.ts` — comment trimmed.

The only canonical URL for group settings is now `/groups/:id/settings`. `nav.system` labelKey stays — it's an internal name, the user-visible label is "Settings" via i18n (renaming the key is a follow-up if you want).

Local: navigation.spec.ts 3/3 chromium passing. Full unit 727/727. typecheck + lint (0 errors) + format clean.

File: frontend/src/app/router.tsx [hunk 4]
```diff
@@ -225,8 +214,6 @@ export function AppRoutes() {
           <Route path="/files/*" element={<UngroupedRedirect />} />
           <Route path="/exports" element={<UngroupedRedirect />} />
           <Route path="/exports/*" element={<UngroupedRedirect />} />
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply frontend/src/app/router.tsx:216]: It's fine. No legacy!

File: go/cmd/inventario/run/bootstrap/config.go [hunk 5]
```diff
@@ -56,11 +56,13 @@ type Config struct {
...
-	// until this flag flips on. The flag is removed once the feature is
-	// fully launched (see §8 in #202).
-	FeatureCurrencyMigration bool `yaml:"feature_currency_migration" env:"FEATURE_CURRENCY_MIGRATION" env-default:"false"`
+	// requireGroupNotMigrating lock middleware, and the worker.
+	// Default on now that the feature shipped end-to-end under #1604 —
+	// flipping it off keeps the schema + registries inert (the kill-switch
+	// path) without rebuilding. Helm's `features.currencyMigration` value
+	// owns the operator-facing toggle. The flag is removed entirely once
+	// the rollout settles (see §8 in #202).
+	FeatureCurrencyMigration bool `yaml:"feature_currency_migration" env:"FEATURE_CURRENCY_MIGRATION" env-default:"true"`
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply go/cmd/inventario/run/bootstrap/config.go:65]: Updated the PR description with a dedicated **⚠️ Operator-visible behavior change** section that calls out:

- The `FeatureCurrencyMigration` env-default flip (`false` → `true`).
- The new `features.currencyMigration` Helm value (default `true`) and the configmap wiring that fans out to every deployment via `envFrom`.
- The mirrored `INVENTARIO_RUN_FEATURE_CURRENCY_MIGRATION: "true"` lines in `k8s/dev/inventario/configmap.yaml` + `k8s/prod/configmap.yaml`.
- The workaround removals from `docker-compose.e2e.yaml`, `.github/workflows/e2e-tests.yml`, and `e2e/setup/setup-stack.ts`.
- Explicit upgrade impact (operators who want the surface dark must now opt-out).

**Comment by denisvmedia** [comment]
CI on ef4a85c was down to **one** failing spec on firefox + webkit (chromium passed):

```
no-group-redirect.spec.ts > NoGroupView drives group creation and returns the user to /
  Expected: /\/g\/mock-slug-1261|\/$/
  Received: http://localhost:3333/no-group?g=mock-slug-1261
```

Race: `NoGroupPage`'s submit awaits the create mutation (whose `onSuccess` awaits `invalidateQueries` on groups + auth/me, so by `mutateAsync`'s return both refetches have settled), then calls `navigate("/")`. Between the invalidate and the navigate, GroupContext's auto-set effect re-runs because `groups` flipped from `[]` to `[mockGroup]` — firefox + webkit's React-scheduler interleaves the `setSearchParams` (replace, `/no-group?g=mock-slug-1261`) ahead of the `navigate`, and the `navigate("/")` somehow doesn't make it through. URL settles at `/no-group?g=mock-slug-1261` and stays for 12+ polls. Chromium's scheduler ordering is the other way and the `navigate` wins, so chromium passes.

Pushed f039c95 — drop `/no-group` from `PATHS_WITH_GROUP_QUERY`. `/no-group` is the zero-group state by definition (no active group to pin), and the transient post-create case is exactly the race above. Comment block in `GroupContext.tsx` spells out the reasoning so a future reader doesn't add it back.

Local: full unit suite **740/740**, typecheck + lint (0 errors) + format clean. CI should now go green across all three browsers.

Also rebased onto current master (the commits got new SHAs as a result; same content), and updated the PR description with a dedicated **⚠️ Operator-visible behavior change** section per @copilot-pull-request-reviewer's feedback on the FeatureCurrencyMigration default flip.
